docs/content/developer/cryptography/on-chain/groth16.mdx

```diff
@@ -83,37 +112,53 @@ fn main() {
+The API in IOTA for verifying a proof expects a special processed verification key, where only a subset of the values are used. Ideally, computation for this prepared verification key happens only once per circuit. You can perform this processing using the `iota::groth16::prepare_verifying_key` method of the IOTA Move API with a serialization of the `params.vk` value used previously. Remember that you should not use this in a production environment, as you should grab a safer entropy resource i.e., from a ceremony.
 
 The output of the `prepare_verifying_key` function is a vector with four byte arrays, which corresponds to the `vk_gamma_abc_g1_bytes`, `alpha_g1_beta_g2_bytes`, `gamma_g2_neg_pc_bytes`, `delta_g2_neg_pc_bytes`.
 
 To verify a proof, you also need two more inputs, `public_inputs_bytes` and `proof_points_bytes`, which contain the public inputs and the proof respectively. These are serializations of the `inputs` and `proof` values from the previous example, which you can compute in Rust as follows:
 
 ```rust
+// Print verifying key
 let mut vk_bytes = Vec::new();
 params.vk.serialize_compressed(&mut vk_bytes).unwrap();
+println!("Verifying key: {}", hex::encode(vk_bytes));
 
+// Print public inputs. Note that they are concatenated.
 let mut public_inputs_bytes = Vec::new();
 for i in 0..inputs.len() { // if there is more than one public input, serialize one by one
     inputs[i].serialize_compressed(&mut inputs_bytes).unwrap();
 }
+println!("Public inputs: {}", hex::encode(public_inputs_serialized));
 
+// Print proof
 let mut proof_points_bytes = Vec::new();
 proof.serialize_compressed(&mut proof_points_bytes).unwrap();
+println!("Proof: {}", hex::encode(proof_serialized));
+```
+
+You can also use snarkjs to verify the generated and exported proof.
+```bash
+snarkjs groth16 verify verification_key.json public.json proof.json
 ```
 
-The following example smart contract prepares a verification key and verifies the corresponding proof. This example uses the BN254 elliptic curve construction, which is given as the first parameter to the `prepare_verifying_key` and `verify_groth16_proof` functions. You can use the `bls12381` function instead for BLS12-381 construction.
+The following example smart contract prepares a verification key and verifies the corresponding proof. This example uses the BN254 elliptic curve construction, which is given as the first parameter to the `prepare_verifying_key` and `verify_groth16_proof` functions. You can use the [`bls12381`](https://docs.iota.org/references/framework/iota/groth16#iota_groth16_bls12381) function instead, as bn254 is not considered safe enough anymore.
```

Comment:
- Please use standard curve names BLS12-381 and BN254.
- BLS12-381 is a curve, not a function.
- This security concern has already been mentioned above, no need to state that again IMO.
- Why not rewrite the example using BLS12-381 directly, and say that users can use a less secure curve BN254?